sql/core/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala

```diff
@@ -242,16 +242,7 @@ class KeyValueGroupedDataset[K, V] private[sql](
   def mapGroupsWithState[S: Encoder, U: Encoder](
       func: (K, Iterator[V], GroupState[S]) => U): Dataset[U] = {
     val flatMapFunc = (key: K, it: Iterator[V], s: GroupState[S]) => Iterator(func(key, it, s))
-    Dataset[U](
-      sparkSession,
-      FlatMapGroupsWithState[K, V, S, U](
-        flatMapFunc.asInstanceOf[(Any, Iterator[Any], LogicalGroupState[Any]) => Iterator[Any]],
-        groupingAttributes,
-        dataAttributes,
-        OutputMode.Update,
-        isMapGroupsWithState = true,
```

Comment:
This change is not correct. It will set `isMapGroupsWithState` to `false`. However, looks like we don't have a unit test covering this :(